dev-lang/rust/rust-1.74.1-r101.ebuild

```diff
@@ -648,13 +655,64 @@ mrustc_bootstrap() {
 	# These flags are used in every invocation of our bootstrap `cargo`.
 	local cargo_flags="--target ${CFG_COMPILER_HOST_TRIPLE} -j $(makeopts_jobs) --release --verbose"
 
+	# for bootstrap, let's using the built-in stdlib of compiler (could be the bundled one)
+	filter-flags '-stdlib=*'
+
+	# mrustc requires gcc, so disable libcxx to avoid linker failure on w/o '-lstdc++'
+	[[ "${LLVM_USE_LIBCXX}" == "1" ]] && unset LLVM_USE_LIBCXX
+
+	local llvm_config_wrapper_cxxflags=0
+
 	if use system-llvm; then
 		export LLVM_CONFIG="$(get_llvm_prefix)/bin/llvm-config"
+
+		local llvm_config_cxxflags=$(${LLVM_CONFIG} --cxxflags)
+		elog "Checking llvm-config --cxxflags: '${llvm_config_cxxflags}'"
+		[[ "${llvm_config_cxxflags}" =~ (^|[[:space:]])-stdlib=libc\+\+([[:space:]]|$) ]] && {
+			elog "Found LLVM CXXFLAGS has \"--stdlib=libc++\""
+			llvm_config_wrapper_cxxflags=1
+		}
 	else
 		llvm_bootstrap
 		export LLVM_CONFIG="${WORKDIR}/bootstrap/llvm/bin/llvm-config"
 	fi
 
+	elog "LLVM_CONFIG before wrappers: ${LLVM_CONFIG}"
+
+	# workaround for gcc bug 122409 on musl by wrapping llvm-config
+	# to append libc++ header if has "-stdlib=libc++"
+	elog "Preparing wrapper of llvm-config (${WORKDIR}/llvm-config)"
+	cat > ${WORKDIR}/llvm-config <<-EOF || die
+	#!/bin/bash
+
+	RULES=()
+	for flag in "\$@"; do
+	    case "\${flag}" in
+	$([[ "${llvm_config_wrapper_cxxflags}" == 1 ]] && {
```

Comment:
Please hoist this outside -- i.e. only create the wrapper and export `LLVM_CONFIG` to be the wrapper if llvm_config_wrapper_cxxflags is 1, don't do it _inside_ the wrapper.

That way, we minimise risk and changes to the usual path.

Reply:
Thanks. I understand now!